File: src/main/java/com/afropreneurdigital/blog/model/Article.java
package com.afropreneurdigital.blog.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.sql.Blob;
import java.util.Date;
import java.util.Set;

@Data
@Entity
@Table(name = "ARTICLE")
public class Article {
    @Id
    @Column(name = "ID", nullable = false, unique = true)
    private Long id;
    private String title;
    private String content;

    @ManyToOne
    @JoinColumn(name = "AUTHOR_ID")
    private Author author;
    private Date releaseDate;
    private Date deletedAt;

    @ManyToMany
    @JoinTable(
            name = "ARTICLE_ENTITY_COMP",
            joinColumns = @JoinColumn(name = "ARTICLE_ID"),
            inverseJoinColumns = @JoinColumn(name = "ENTITY_ID"))
    private Set<Entities> topics;

    public Article() {}
}

File: src/main/ui/src/app/model/Article.ts
export interface Article {
  id: number;
  title: string;
  releaseDate: string;
  author: Author;
  topics: Topic[];
  content: string;
}

export interface Topic {
  id: number;
  group: string;
  key: string;
  value: string;
}

export interface Author {
  id: number;
  fullName: string;
}
